db/memtable.cc

```diff
@@ -725,8 +728,8 @@ Status MemTable::Add(SequenceNumber s, ValueType type,
     }
   }
 
-  size_t ts_sz = GetInternalKeyComparator().user_comparator()->timestamp_size();
-  Slice key_without_ts = StripTimestampFromUserKey(key, ts_sz);
+  MaybeUpdateNewestUDT(key_slice);
```

Comment:
You might only be able to do this in `Add()` in the `!allow_concurrent` branch. I know you're already doing it here, but a race condition detector might complain about it at some point. The convention for `allow_concurrent` appears to be to apply updates requiring serialization in `BatchPostProcess()`

Reply:
Thank you for pointing this out! I have for now move it to only the !allow_concurrent branch and added a TODO for it. Looks like all the member variables that `BatchPostProcess()` updates have built-in thread safety support via std::atomic, so currently its invocation is not specifically serialized. A hacky workaround would be to track the newest UDT as uint64_t here since we are only supporting that type of user-defined timestamp for this feature already. Anyways, I mentioned this caveat in the feature description and will follow up with MyRocks on the priority of having this.